pkg/routeagent/main.go

```diff
@@ -37,6 +37,8 @@ func filterRouteAgentPods(options *v1.ListOptions) {
 	options.LabelSelector = route.SmRouteAgentFilter
 }
 
+const resyncInterval = time.Hour * 24
```

Comment:
24 hours is essentially useless. I think we should just set it to 0 to disable it. As long as we're re-queueing appropriately in `processNextEndpoint` and `processNextPod` on failures then we shouldn't need this.

Reply:
Yeah, may be it's even something I should add to be configurable, for just in case it has unintended consequences... as I told @sridhargaddam on the leader-election patch. Sridhar, what do you think?.

Reply:
IMHO, a resync interval of 24 hours is reaallyyy a long time and I don't see any value in having it. If we want to increase the current value of 30 secs, we could use something close (like in minutes) or disable it all-together as @tpantelis suggested (assuming that we are properly handling the re-queing event). 

Regarding making it configurable params, TBH I'm not a big fan of introducing too many configurable params which would affect the user-experience. I'd rather prefer to use some reasonable defaults where applicable and only create configurable params where it really matters. My 2 cents.

Reply:
Yeah, but the parameters, with sensible defaults are useful when problems arise on production, you can mitigate instead of keeping a customer waiting for a patch which could take some time to produce and ship.

I'm not a big fan of parameters or asking the user for input. I think it's fine if all this is hidden to the normal user via subctl, etc...

Reply:
The resync seems problematic in `processNextEndpoint`. Every period we'll create a new vxlan interface, w/o deleting the old one as we do in `handleRemovedEndpoint`.  We would need to make `processNextEndpoint` idempotent.  It is intended to re-process in case something went wrong previously and may not have been re-queued but we haven't seen a case where that would come into play. So I don't think we need auto resync at all and, in fact, it seems it may be detrimental. Unless I'm missing something?